lib/libzutil/zutil_import.c

```diff
@@ -929,22 +1006,35 @@ zpool_read_label(int fd, nvlist_t **config, int *num_labels)
 
 	if (lio_listio(LIO_WAIT, aiocbps, VDEV_LABELS, NULL) != 0) {
 		int saved_errno = errno;
-
-		if (errno == EAGAIN || errno == EINTR || errno == EIO) {
-			/*
-			 * A portion of the requests may have been submitted.
-			 * Clean them up.
-			 */
-			for (l = 0; l < VDEV_LABELS; l++) {
-				errno = 0;
-				int r = aio_error(&aiocbs[l]);
-				if (r != EINVAL)
+		int tmpret = -1;
+
+		for (l = 0; l < VDEV_LABELS; l++) {
+			errno = 0;
+			int r = aio_error(&aiocbs[l]);
+			if (r != 0) {
+				if (r == EOPNOTSUPP) {
+					/*
```

Comment:
You still need to call `aio_return` here for all operations.  Otherwise you'll leak AIO resources.

Reply:
Sure.

If I'm understanding this correctly, doesn't the need to always call aio_return imply that the old code also leaked resources if you got back something that wasn't (EINTR|EAGAIN|EIO)? (I'm finding it hard to find documentation on when it's appropriate to call versus not - the closest thing I found was a macOS man page which said you should always do it...

Reply:
No.  You only need to call `aio_return` if the aiocb was actually started.